tbx/core/tests/test_util_tags.py

```diff
@@ -0,0 +1,77 @@
+from datetime import datetime, time
+
+from django.test import SimpleTestCase
+
+from tbx.core.templatetags.util_tags import format_date_for_event
+
+
+class TestFormatDateForEvent(SimpleTestCase):
+    def test_with_start_date_only(self):
+        self.assertEqual(
+            format_date_for_event(datetime(2021, 1, 1).date(), None, None, None),
+            "01 Jan 2021",
+        )
+
+    def test_with_start_date_and_start_time(self):
+        self.assertEqual(
+            format_date_for_event(datetime(2021, 1, 1).date(), time(13, 0), None, None),
+            "01 Jan 2021, 1pm",
+        )
+
+    def test_with_start_date_and_end_date(self):
+        self.assertEqual(
+            format_date_for_event(
+                datetime(2021, 1, 1).date(), None, datetime(2021, 1, 2).date(), None
+            ),
+            "01 Jan 2021 - 02 Jan 2021",
+        )
+
+    def test_with_start_date_and_start_time_and_end_date_and_end_time_on_same_day(self):
+        self.assertEqual(
+            format_date_for_event(
+                datetime(2021, 1, 1).date(),
+                time(13, 0),
+                datetime(2021, 1, 1).date(),
+                time(14, 0),
+            ),
+            "01 Jan 2021, 1-2pm",
+        )
+
+    def test_with_start_date_and_start_time_and_end_date_on_same_day_with_different_time_suffix(
+        self,
+    ):
+        self.assertEqual(
+            format_date_for_event(
+                datetime(2021, 1, 1).date(),
+                time(11, 0),
+                datetime(2021, 1, 1).date(),
+                time(14, 0),
+            ),
+            "01 Jan 2021, 11am-2pm",
+        )
+
+    def test_with_start_date_and_start_time_and_end_date_and_end_time_on_different_day(
+        self,
+    ):
+        self.assertEqual(
+            format_date_for_event(
+                datetime(2021, 1, 1).date(),
+                time(13, 0),
+                datetime(2021, 1, 2).date(),
+                time(14, 0),
+            ),
+            "01 Jan 2021, 1pm - 02 Jan 2021, 2pm",
+        )
+
+    def test_with_start_date_and_start_time_and_end_date_on_different_day_with_different_time_suffix(
+        self,
+    ):
+        self.assertEqual(
+            format_date_for_event(
+                datetime(2021, 1, 1).date(),
+                time(11, 0),
+                datetime(2021, 1, 2).date(),
+                time(14, 0),
+            ),
+            "01 Jan 2021, 11am - 02 Jan 2021, 2pm",
+        )
```

Comment:
**Question**: Is it possible to set a start date but no start time, and an end date and end time on a different day? E.g. "01 Jan 2021 - 02 Jan 2021, 2pm"?

Reply:
Based on the other comment, I was thinking that we should add a condition to make it not possible to add an end time without a start time. But `01 Jan 2021 - 02 Jan 2021, 2pm` also makes sense.

Reply:
End time without a start time doesn't make much sense to me - let's disallow it if possible

Reply:
Added check in cec74fe5cbc7676f1ee282e0d9be7d03c29f33d3